### DISCUSSION TITLE
React Native Native Wind

### DISCUSSION THREAD

**Discussion by jerry-mh** [nativewind/nativewind#1353] [Q&A]
How can I integrate NativeWind into an existing Expo project? I'm facing difficulties because the project I'm following for learning is 6 months old and uses React Native v2, while my project is using React Native v4. Could you guide me on how to set it up correctly?

**Comment by cloud0816** [comment]
Hello. I hope my answer help you.


**1. Ensure Expo and React Native Are Up-to-Date**
`expo upgrade`

**2. Install NativeWind**
Add NativeWind to your project using npm or yarn. Run the following command in your project directory
`npm install nativewind`   or  `yarn add nativewind`

**3. Install React Native Babel Plugin**
NativeWind requires a Babel plugin to function correctly. Install the React Native Babel plugin if it’s not already installed
`npm install --save-dev babel-plugin-react-native-classname-to-style`
or
`yarn add --dev babel-plugin-react-native-classname-to-style`

**4. Configure Babel**
Modify your Babel configuration to include the NativeWind plugin. If you have a `babel.config.js` file, add the following
```
 module.exports = function(api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
    plugins: ['nativewind/babel', 'react-native-classname-to-style']
  };
};
```

**5. Tailwind Configuration**
Create a `tailwind.config.js` file at the root of your project to customize your Tailwind CSS setup
```
 module.exports = {
  content: [
    "./App.{js,jsx,ts,tsx}",
    "./src/**/*.{js,jsx,ts,tsx}"
  ],
  theme: {
    extend: {},
  },
  plugins: [],
};
```

**6. Using NativeWind**
Now you can use Tailwind CSS-style classes in your React Native components. 
For example:
```
import { Text, View } from 'react-native';
import { styled } from 'nativewind';

const StyledView = styled(View);
const StyledText = styled(Text);

export default function App() {
  return (
    <StyledView className="flex-1 items-center justify-center bg-white">
      <StyledText className="text-lg font-bold">
        Hello, NativeWind!
      </StyledText>
    </StyledView>
  );
}
```

**7. Restart Your Development Server**
`expo start`

Thank you.

**Reply by jerry-mh** [reply]
is your answer according to native wind 4

**Reply by cloud0816** [reply]
No. According to NativeWind version 3.

**Reply by jerry-mh** [reply]
iam using native wind version 4

**Comment by cloud0816** [comment]
hi jerry-mh. sorry for late reply.

change this:

step3:
`npm install --save-dev babel-plugin-nativewind` or `yarn add --dev babel-plugin-nativewind`

step4:
```
module.exports = function(api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
    plugins: ['nativewind/babel']
  };
};
```
step5:
```
module.exports = {
  content: [
    "./App.{js,jsx,ts,tsx}",
    "./src/**/*.{js,jsx,ts,tsx}"
  ],
  theme: {
    extend: {},
  },
  plugins: [],
};
```
step6:
```
import { Text, View } from 'react-native';
import { NativeWindProvider } from 'nativewind';

export default function App() {
  return (
    <NativeWindProvider>
      <View className="flex-1 items-center justify-center bg-white">
        <Text className="text-lg font-bold">
          Hello, NativeWind!
        </Text>
      </View>
    </NativeWindProvider>
  );
}
```

please check this. thank you.

**Comment by danstepanov** [comment]
- Nativewind v4 + Expo SDK 53: `npx rn-new@latest --nativewind`
- Nativewind v5 + Expo SDK 54: `npx rn-new@next --nativewind`
- Manual setup for v4: https://www.nativewind.dev/getting-started/installation
- Manual setup for v5: https://www.nativewind.dev/v5/getting-started/installation
